types/fabric/fabric-impl.d.ts

```diff
@@ -3506,8 +3506,19 @@ export class Object {
 	intersectsWithRect(pointTL: any, pointBR: any, absolute?: boolean, calculate?: boolean): boolean;
 	/**
 	 * Animates object's properties
+	 * object.animate('left', ..., {duration: ...});
+	 * @param property Property to animate
+	 * @param value Value to animate property
+	 * @param options The animation options
+	 */
+	animate(property: string, value: number | string, options?: IAnimationOptions): Object;
+	/**
+	 * Animates object's properties
+	 * object.animate({ left: ..., top: ... }, { duration: ... });
+	 * @param properties Properties to animate
+	 * @param value Options object
 	 */
-	animate(): Object;
+	animate(properties: any, options?: IAnimationOptions): Object;
```

Comment:
Ideally it would be nice to have an interface specifying the valid type. Otherwise, `object` would be better than `any` here.

Reply:
Ok, I believe I got the right interface for the parameter. Had to do a little digging through source to make sure, but I believe that [only numbers or strings are _really_ allowed.](https://github.com/fabricjs/fabric.js/blob/master/src/mixins/animation.mixin.js#L161)